site/autoclustering.xml

```diff
@@ -0,0 +1,189 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<html xmlns="http://www.w3.org/1999/xhtml"
+    xmlns:doc="http://www.rabbitmq.com/namespaces/ad-hoc/doc"
+    xmlns:x="http://www.rabbitmq.com/2011/extensions">
+  <head>
+    <title>Cluster Formation</title>
+  </head>
+  <body>
+    <doc:section name="intro">
+      <doc:heading>Introduction</doc:heading>
+
+      <p>
+        A RabbitMQ cluster can formed in a number of ways:
+
+        <ul>
+          <li>Declaratively by listing cluster nodes in <a href="/configure.html">config file</a></li>
+          <li>Declaratively using DNS-based discovery</li>
+          <li>
+            Declaratively using <a href="https://github.com/aweber/rabbitmq-autocluster/">rabbitmq-autocluster</a> (a plugin),
+            which supports AWS peer discovery (tags and autoscaling group membership-based), etcd, Consul, and more backends.
+          </li>
+          <li>Manually with <code>rabbitmqctl</code></li>
+        </ul>
+      </p>
+
+      <p>
+        When a node starts and detects it doesn't have a previously
+        initialised database, it will check if there's a peer
+        discovery mechanism configured. If that's the case, it will
+        then perform the discovery and attempt to contact each
+        discovered peer in order. Finally, it will attempt to join the
+        cluster of the first reachable peer.
+      </p>
+
+      <p>
+        If peer discovery isn't configured, or it fails, or no peers are reachable,
+        the node will initialise from scratch and proceed as a standalone node.
+      </p>
+
+      <p>
+        Nodes that previously were cluster members will try to contact
+        their "last seen" peer for a period of time. In this case, no peer discovery
+        is performed.
+      </p>
+    </doc:section>
+
+    <doc:section name="peer_discovery">
+      <doc:heading>Peer Discovery</doc:heading>
+
+      <p>
+        To form a cluster, new ("blank") nodes need to be able to discover their
+        peers. As of RabbitMQ 3.7.0, peer discovery mechanisms are pluggable.
+        There are two implementations available out of the box:
+
+        <ul>
+          <li>Config file-based (a.k.a. the "classic config" backend)</li>
+          <li>DNS-based</li>
+        </ul>
+      </p>
+      <p>
+        The discovery mechanism to use is specified in the <a
+        href="/configure.html">config file</a>, as are
+        mechanism-specific settings, for example, discovery service
+        hostnames, credentials, and so on.
+        <code>autocluster.peer_discovery_backend</code> is the key
+        that controls what discovery module (implementation) is used.
+        The module has to implement the <a
+        href="https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_peer_discovery.erl">rabbit_peer_discovery</a>
```

Comment:
It should be `rabbit_peer_discovery_backend` at https://github.com/rabbitmq/rabbitmq-common/blob/master/src/rabbit_peer_discovery_backend.erl